backend/app/api/v1/commons/utils.py

```diff
@@ -160,9 +160,13 @@ def get_dict_from_qs(qs):
 
 def construct_query(filter_dict):
     query_parts = []
+
     if isinstance(filter_dict, dict):
         for key, values in filter_dict.items():
             k = constants.FIELDS_FILTER_DICT[key]
+            # as status values are mapped differenlty for telco
+            if k == "status":
+                values = [constants.TELCO_STATUS_MAP[val] for val in values]
```

Comment:
Instead of taking an exception if you run into a value that's *not* in the `TELCO_STATUS_MAP` keys, maybe `constants.TELCO_STATUS_MAP.get(val, val)`?